lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/ldap/JdkLdapClient.java

```diff
@@ -94,13 +99,31 @@ public <T> T processLdapContext(String userName, String password, LdapContextPro
     }
 
     @Override
-    public <T> T executeLdapQuery(String userName, String password, LdapQuery ldapQuery, LdapSearchResultProcessor<T> resultProcessor)
+    public synchronized <T> T executeLdapQuery(String userName, String password, LdapQuery ldapQuery, LdapSearchResultProcessor<T> resultProcessor)
```

Comment:
If we add `synchronized` here, it will slow down all the requests to this function. I think it's better to use `ConcurrentHashMap` instead of `HashMap` then `synchronized` lock the object inside the map when using it

Reply:
Make sense. Thanks for advicing!